tests/noto_fonts_for_android_test.py

```diff
@@ -121,5 +121,31 @@ def test_font_weights():
   assert not errors, ", ".join(errors)
 
 
+def test_font_weight_coverage():
+  root = etree.parse(str(_noto_4_android_file()))
+  errors = []
+  reached_max_weight = set()
+  reached_min_weight = set()
+  all_fonts = set()
+  for font_el in root.xpath("//font[@path]"):
+    xml_weight = int(font_el.attrib["weight"])
+    path = _font_path(font_el)
+    all_fonts.add(_font_file(font_el))
+
+    font = _open_font(font_el)
+    min_wght, default_wght, max_weight = _weight(font)
+    if xml_weight == min_wght:
+      reached_min_weight.add(_font_file(font_el))
+    if xml_weight == max_weight:
+      reached_max_weight.add(_font_file(font_el))
+
+  print("======> minimum not reached")
+  for font_file_name in all_fonts.difference(reached_min_weight):
+    print(font_file_name)
+  print("======> maximum not reached")
+  for font_file_name in all_fonts.difference(reached_max_weight):
+    print(font_file_name)
```

Comment:
As written the test doesn't actually fail if you violate it's conditions.

Are you sure you wnat to reach the min/max? In looking at fonts.xml I suspect we want to hit all whole multiple of 100 between min and max weight?

I think that might lead me to write the test as something along the lines of:

```
# at the top
import collections

# in the test
for family in root.iter("family"):
  font_to_xml_weights = collections.defaultdict(set)
  for font in family.xpath("//font[@path]"):  # ...I think you can xpath within a 
    font_to_xml_weights[_font_path(font)).add(int(font_el.attrib["weight"]))

  # now you have a map of font path => set of weights in xml
  for font_path, xml_weights in font_to_xml_weights.items():
    # open the font, compute the 100 weights between it's min/max weight
   # if xml_weights != computed weights add this to the error list

  assert there are no errors, ", ".join(errors)
```

Reply:
Your code looks better. In my code, I wanted to check if we ever have this situation (just in case this omission was intentional; not exposing all the weights is not wrong if done purposefully [I have no idea for what purpose though :-)]). BTW, SansOriya and SansOriyaUI are in this situation and I will expand the XML to cover the whole range in these VFs.

Reply:
You could have a list of expected exceptions to the rule perhaps? - that way it'll flag if, as I believe is normal, it's a mistake and we still have an opt-out.

I'm curious ... when would we want to NOT list all the weights?